Expose bpf-lb-sock-hostns-only in cilium status

romanspb80 opened the pull request:
BPFSocketLBHostnsOnly is outputted in the Kube Proxy Replacement section in cilium status --verbose:

$ cilium status --verbose
[...]
KubeProxyReplacement Details:
  [...]
  BPFSocketLBHostnsOnly:   Enabled
  [...]

Fixes: #24160

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/master/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

aditighag commented on api/v1/models/kube_proxy_replacement.go:
```diff
@@ -43,6 +43,9 @@ type KubeProxyReplacement struct {
 	Mode string `json:"mode,omitempty"`
+
+	// flag bpf-lb-sock-hostns-only
+	BPFSocketLBHostnsOnly bool `json:"bpfSocketLBHostnsOnly,omitempty"`
```
This should go under `KubeProxyReplacementFeatures` (see above L 41). Please take a look at how other flags are exposed.

romanspb80 replied:
Thanks. I've added the flag to the openapi

aditighag replied:
This is still not addressed. You've added the flag under `KubeProxyReplacement`, and not `KubeProxyReplacementFeatures`. Was there an issue in adding it under the latter?

borkmann commented on pkg/client/client.go:
```diff
@@ -633,6 +638,7 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 		fmt.Fprintf(tab, "  Status:\t%s\n", sr.KubeProxyReplacement.Mode)
 		fmt.Fprintf(tab, "  Socket LB:\t%s\n", socketLB)
 		fmt.Fprintf(tab, "  Socket LB Tracing:\t%s\n", socketLBTracing)
+		fmt.Fprintf(tab, "  BPFSocketLBHostnsOnly:\t%s\n", bpfSocketLBHostnsOnly)
```
nit, this should better say:

`Socket LB Coverage:` and then spill out either `Hostns-only` or `Full` for example.

romanspb80 replied:
Ok. Done.

borkmann replied:
Looks great, thanks @romanspb80 . Could you still add your Signed-off-by to the commit message? If you look intothe git log, you'll see that all commits have it. Then we should be good to move forward and merge. Thanks!

aditighag reviewed (approved):
Thanks for your contribution. LGTM... This is ready for merge once you add "Signed-off-by:" in the commit message.

aditighag reviewed (changes requested):
https://github.com/cilium/cilium/pull/24570#discussion_r1153562110

romanspb80 committed:
Expose bpf-lb-sock-hostns-only in cilium status

BPFSocketLBHostnsOnly is outputted in the Kube Proxy Replacement section in cilium status --verbose:

$ cilium status --verbose
[...]
KubeProxyReplacement Details:
  [...]
  Socket LB Coverage:     Hostns-only
  [...]

Fixes: #24160

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>

aditighag reviewed (approved):
Thanks!
Integration test failure is same as - https://github.com/cilium/cilium/issues/24653. 
The PR is ready to merge.
